src/main/java/core/mate/academy/service/MachineProducer.java

```diff
@@ -1,11 +1,13 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
 import java.util.List;
 
-public interface MachineProducer {
+public interface MachineProducer<T extends Machine> {
```

Comment:
Don't begin class or method implementation with an empty line. The interface declaration should be followed by the method declaration without an empty line in between.